### DISCUSSION TITLE
Unexpected behaviour during save with associations

### DISCUSSION THREAD

**Discussion by thorque** [sequelize/sequelize#16698] [Help & Questions]
Hello,

I've tried to store a new entity with an association added:
```javascript
const author = models.author.build({
  firstName: faker.person.firstName(), 
  lastName: faker.person.lastName()
});
await author.save();

const post = models.post.build({
  title: faker.lorem.sentence(),
  content: faker.lorem.paragraph()
})
post.setAuthor(author);
await post.save();
``` 

With this code this sql will be executed:
`INSERT INTO "posts" ("authorId","updatedAt","createdAt") VALUES ($1,$2,$3);` 

This is a little bit unexpected that the values for title and content are ignored. Due to the not null constraint an error occurs.

Does sequelize not support this type of saving? 

Best
Thorsten

**Comment by thorque** [comment]
With 2 separates post.save() it works properly:

```javascript
await post.save();
post.setAuthor(author);
await post.save();
```

**Comment by ephys** [comment]
`setAuthor` is an async method that immediately saves, you need to pass `{ save: false }` to not immediately save on `setAuthor`
